src/scripts/generate_binary_ops.py

```diff
@@ -0,0 +1,325 @@
+            expression += (" %s %s)" % (result["oper"], self.TEST_VALUE_VAR_NAME.format(NUMBER = (i + 2))))
+            if result["oper"] == "/":
+                expression += " | 0)"
+                expression = "(" + expression
+            self.operands[i + 1] = result["op2"]
+            last = result["sum"]
+        expression += "\n"
+        expected_value = last
+        # Generating false numbers
+        false_numbers = self.generate_false_numbers(expected_value, _FALSE_VALUES_COUNT, MINNUM, MAXNUM)
+        # Creating and adding false numbers string
+        false_numbers_string = self.create_false_numbers_array_string(_FALSE_VALUES_COUNT)
+        # Creating and adding declarations of test values
+        test_case = self.create_declarations_string(self.create_declarations_dict(self.operands,
+        self.TEST_VALUE_VAR_NAME))
+        # Creating and adding declarations of false values
+        test_case += self.create_declarations_string(self.create_declarations_dict(false_numbers,
+        self.FALSE_RESULT_VAR_NAME))
+        # Creating and adding declarations of expected value
+        test_case += self.create_declarations_string(self.create_declarations_dict(expected_value,
+        self.EXPECTED_VAR_NAME))
+        # Creating and adding the function call
+        test_case += self.create_function_call(expression, self.EXPECTED_VAR_NAME.format(NUMBER = ""),
+        false_numbers_string, validate.validate_numeric_header)
+        return test_case
+
+    # Randomly generates the type of the expression out of the needed calculation
+    def actual_calc(self, arithcalc, bitcalc, urscalc):
+        # 1: arithmetic, 2: bitwise (excluding unsigned right shift) 3: unsigned right shift
+        if arithcalc and not bitcalc and not urscalc: # If we need bitwise, arithmetic calculations
+                                                        # or unsigned right shift
+            calc = _ARITH
+        elif bitcalc and not arithcalc and not urscalc:
+            calc = _BIT
+        elif urscalc and not arithcalc and not bitcalc:
+            calc = _UINT32
+        elif arithcalc and bitcalc and not urscalc:
+            calc = random.randint(1, 2) # 1 or 2 (_ARITH or _BIT)
+        elif bitcalc and urscalc and not arithcalc:
+            calc = random.randint(2, 3) # 2 or 3 (_BIT or _UINT32)
+        elif arithcalc and urscalc and not bitcalc:
+            calc = (1 + random.randint(0, 1) * 2) # Must be 1 or 3 (_ARITH or _UINT32)
+        elif arithcalc and bitcalc and urscalc:
+                calc = random.randint(1, 3) # 1, 2 or 3 (Could be any of them)
+        return calc
+
+    # Calculating currently needed calculation types
+    def needed_calculation_types(self):
+        arithcalc = False
+        bitcalc = False
+        urscalc = False
+        for op in self.options.test_cases:
+            # If arithmetic operators are currently needed
+            if op in _ARITHMETIC_OPERATORS:
+                arithcalc = True
```

Comment:
I see your point here, but I can not simplify these 2 lines to 1 line, because if it is false, I must not change the value, or do you mean to change only the last line to this suggestion? (The same question for the other 2 below)